### DISCUSSION TITLE
Shap Characterstic Function

### DISCUSSION THREAD

**Discussion by sourabh181818** [shap/shap#2945] [Q&A]
So, in the equation below
![SHAP Equation](https://github.com/slundberg/shap/assets/118224119/fa58272b-4468-4ae3-934a-e7d9b0b0a6d5)
The left part in front of the summation symbol represents how we weigh the difference between the model with the feature of interest and the model without the feature of interest. The part to its right talks about this difference. As we cannot train the model every time for various combinations of subsets and feature sets, SHAP does this by mapping a characteristic function for these subsets. What is this characteristic function? And what is the equation to calculate it if I am supposed to calculate SHAP values manually?

**Comment by jainraj** [comment]
That characteristic function is the whole game here! See [Algorithms to estimate Shapley value feature attributions - Chen et al.](https://arxiv.org/pdf/2207.07605.pdf) - especially Section 4.
